### PR TITLE
SecurityPkg: Additional CodeQL fixes

### PR DESCRIPTION
**Pull request by TaylorBeebe**
## Description

Various fixes.

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [x] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Build SecurityPkg and boot changes on QemuQ35Pkg to EFI shell.

## Integration Instructions

N/A

### CODE REVIEW

**Review by kenlautner — APPROVED** [review]

File: SecurityPkg/FvReportPei/FvReportPei.c [hunk 1]
```diff
@@ -352,12 +372,17 @@ CheckStoredHashFv (
...
-    Status   = VerifyHashedFv (
+    // MU_CHANGE [START] - CodeQL change
+    if (HashInfo != NULL) {
+      Status = VerifyHashedFv (
                  HashInfo,
                  StoredHashFvPpi->FvInfo,
                  StoredHashFvPpi->FvNumber,
                  BootMode
                  );
+    }
```
Comment (kenlautner) [inline SecurityPkg/FvReportPei/FvReportPei.c:385]: If HashInfo is NULL shouldn't we edit Status to be some form of error?  Currently we can have it null and it'll assume the verification was successful.

File: SecurityPkg/Library/AuthVariableLib/AuthService.c [hunk 2]
```diff
@@ -439,7 +442,14 @@ CheckSignatureListFormat (
   // Walk through the input signature list and check the data format.
   // If any signature is incorrectly formed, the whole check will fail.
   //
-  while ((SigDataSize > 0) && (SigDataSize >= SigList->SignatureListSize)) {
+  // MU_CHANGE [START] - CodeQL change
+  while (SigDataSize > 0) {
+    SigListSize = (UINTN)SigList->SignatureListSize;
```
Comment (os-d) [inline SecurityPkg/Library/AuthVariableLib/AuthService.c]: nit: this seems fine, but the smaller change, like the two below, would be to just do the cast in the while loop, right?

File: SecurityPkg/Tcg/TcgConfigDxe/TcgConfigImpl.c [hunk 3]
```diff
@@ -193,8 +193,15 @@ TcgExtractConfig (
...
-    ConfigRequest    = AllocateZeroPool (Size);
+    // MU_CHANGE [START] - CodeQL change
+    if (ConfigRequestHdr == NULL) {
+      ASSERT (ConfigRequestHdr != NULL);
+      return EFI_OUT_OF_RESOURCES;
+    }
+
+    // MU_CHANGE [END] - CodeQL change
+    Size          = (StrLen (ConfigRequestHdr) + 32 + 1) * sizeof (CHAR16);
+    ConfigRequest = AllocateZeroPool (Size);
```
Comment (os-d) [inline SecurityPkg/Tcg/TcgConfigDxe/TcgConfigImpl.c:203]: Seems like CodeQL missed ConfigRequest not getting a null check?

File: SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c [hunk 4]
```diff
@@ -3474,8 +3500,15 @@ SecureBootExtractConfig (
...
-    ConfigRequest    = AllocateZeroPool (Size);
+    // MU_CHANGE [START] - CodeQL change
+    if (ConfigRequestHdr == NULL) {
+      ASSERT (ConfigRequestHdr != NULL);
+      return EFI_OUT_OF_RESOURCES;
+    }
+
+    // MU_CHANGE [END] - CodeQL change
+    Size          = (StrLen (ConfigRequestHdr) + 32 + 1) * sizeof (CHAR16);
+    ConfigRequest = AllocateZeroPool (Size);
```
Comment (os-d) [inline SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c:3510]: Seems like CodeQL missed ConfigReuest not being null checked?

**Review by os-d — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

**Pull request merged**